runtime/caml/fail.h

```diff
@@ -27,6 +27,9 @@
 #include "mlvalues.h"
 
 #ifdef CAML_INTERNALS
+// Builtin exceptions stored in the bytecode.
+// All of these exceptions are fields in caml_global_data object,
+// which is stored in the bytecode in the DATA section.
```

Comment:
```suggestion
/* Built-in exceptions. In bytecode, these exceptions are the first fields in
   caml_global_data (which is loaded from the bytecode DATA section) - see
   bytecomp/bytelink.ml. In native code, these exceptions are created if
   needed in the startup object - see asmcomp/asmlink.ml and
   Cmm_helpers.predef_exception. */
```

Reply:
I didn't mention the native code initially, because these defines are only used in the bytecode, but I guess it makes sense to document it here. Changed the comment.